src/Sylius/Behat/Context/Api/Admin/ManagingPromotionCouponsContext.php

```diff
@@ -0,0 +1,467 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Behat\Context\Api\Admin;
+
+use Behat\Behat\Context\Context;
+use Sylius\Behat\Client\ApiClientInterface;
+use Sylius\Behat\Client\ResponseCheckerInterface;
+use Sylius\Behat\Context\Api\Resources;
+use Sylius\Behat\Service\Converter\SectionAwareIriConverterInterface;
+use Sylius\Component\Core\Model\PromotionCouponInterface;
+use Sylius\Component\Core\Model\PromotionInterface;
+use Webmozart\Assert\Assert;
+
+final class ManagingPromotionCouponsContext implements Context
+{
+    public function __construct(
+        private ApiClientInterface $client,
+        private ResponseCheckerInterface $responseChecker,
+        private SectionAwareIriConverterInterface $sectionAwareIriConverter,
+    ) {
+    }
+
+    /**
+     * @Given /^I am browsing coupons of (this promotion)$/
+     * @When /^I want to view all coupons of (this promotion)$/
+     */
+    public function iWantToViewAllCouponsOfThisPromotion(PromotionInterface $promotion): void
+    {
+        $this->client->index(Resources::PROMOTION_COUPONS);
+        $this->client->addFilter(
+            'promotion',
+            $this->sectionAwareIriConverter->getIriFromResourceInSection($promotion, 'admin'),
+        );
+        $this->client->filter();
```

Comment:
This looks like a perfect place to use a subresource instead of a filter 🤔

Reply:
Currently, it's not possible to add filters to the subresource, so I kept these filters. Also, I added a new endpoint for the subresource and used it wherever I could.